src/test/xml_coverage.go

```diff
@@ -38,14 +42,108 @@ func parseXMLLines(lines []xmlCoverageLine) []core.LineCoverage {
 	return ret
 }
 
+// Covert the Coverage result to XML bytes, ready to be write to file
+func CoverageResultToXML(sources []core.BuildLabel, coverage core.TestCoverage) []byte {
```

Comment:
nit: this probably shouldn't be exported (it is fine for it to be a separate function, but not sure anything outside the package would want this)